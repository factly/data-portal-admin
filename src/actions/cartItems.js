import axios from '../utils/axios';
import {
  ADD_CARTITEMS,
  SET_CARTITEM_LOADING,
  SET_CARTITEM_REQUEST,
  SET_CARTITEM_IDS,
  CARTITEM_API,
} from '../constants/cartItems';
import { addProducts } from './products';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadCartItems = (id, page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      cartItems: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setCartItemIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${CARTITEM_API(id)}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { id, page: page, limit: limit, ids: currentPageIds };
    dispatch(setCartItemRequest(currentReq, total));
    dispatch(addCartItems(nodes));
    dispatch(setCartItemIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

const setLoading = (loading) => {
  return {
    type: SET_CARTITEM_LOADING,
    payload: { loading },
  };
};

export const addCartItems = (cartItems) => (dispatch) => {
  const products = getValues(cartItems, 'product');
  dispatch(addProducts(products));

  dispatch({
    type: ADD_CARTITEMS,
    payload: {
      cartItems: buildObjectOfItems(deleteKeys(cartItems, ['product'])),
    },
  });
};

const setCartItemRequest = (req, total) => {
  return {
    type: SET_CARTITEM_REQUEST,
    payload: { req, total },
  };
};

const setCartItemIds = (ids) => {
  return {
    type: SET_CARTITEM_IDS,
    payload: { ids },
  };
};
