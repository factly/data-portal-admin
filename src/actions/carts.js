import axios from '../utils/axios';
import {
  baseUrl,
  LOADING_CARTS,
  LOADING_CART_DETAILS,
  LOAD_CARTS_SUCCESS,
  LOAD_CARTS_FAILURE,
  GET_CART_ITEMS_SUCCESS,
  GET_CART_ITEMS_FAILURE,
} from '../constants/carts';

export const loadCarts = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCarts());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadCartsFailure(error.message));
    });

    if (response) {
      dispatch(loadCartsSuccess(response.data));
    }
  };
};

export const getCartItems = (id, page, limit) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}/items`;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCartDetails());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(getCartItemsFailure(error.message));
    });

    if (response) {
      dispatch(getCartItemsSuccess(response.data));
    }
  };
};

const loadingCarts = () => {
  return {
    type: LOADING_CARTS,
  };
};

const loadingCartDetails = () => {
  return {
    type: LOADING_CART_DETAILS,
  };
};

const loadCartsSuccess = (data) => {
  return {
    type: LOAD_CARTS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const loadCartsFailure = (message) => {
  return {
    type: LOAD_CARTS_FAILURE,
    payload: message,
  };
};

const getCartItemsSuccess = (data) => {
  return {
    type: GET_CART_ITEMS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const getCartItemsFailure = (message) => {
  return {
    type: GET_CART_ITEMS_FAILURE,
    payload: message,
  };
};
