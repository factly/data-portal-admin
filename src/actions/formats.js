import axios from 'axios';
import {
  ADD_FORMAT,
  ADD_FORMATS,
  SET_FORMAT_LOADING,
  SET_FORMAT_REQUEST,
  SET_FORMAT_IDS,
  RESET_FORMAT,
  FORMAT_API,
} from '../constants/formats';
import { getIds, buildObjectOfItems } from '../utils/objects';

export const loadFormats = (query) => {
  return async (dispatch, getState) => {
    dispatch(setLoading(true));

    const response = await axios({
      url:FORMAT_API,
      method: 'get',
      params: query
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { ...query, ids: currentPageIds };
    dispatch(setFormatRequest(currentReq, total));
    dispatch(addFormats(nodes));
    dispatch(setFormatIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

export const createFormat = (data) => {
  return async (dispatch, getState) => {
    dispatch(setLoading(true));

    await axios({
      url: FORMAT_API,
      method: 'post',
      data: data,
    });

    dispatch(resetFormat());
    dispatch(setLoading(false));
  };
};

export const updateFormat = (id, data) => {
  return async (dispatch, getState) => {
    let url = `${FORMAT_API}/${id}`;

    dispatch(setLoading(true));

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    });

    dispatch(addFormat(response.data));
    dispatch(setLoading(false));
  };
};

export const deleteFormat = (id) => {
  return async (dispatch, getState) => {
    let url = `${FORMAT_API}/${id}`;

    dispatch(setLoading(true));

    await axios({
      url: url,
      method: 'delete',
    });

    dispatch(resetFormat());
    dispatch(setLoading(false));
  };
};

export const getFormat = (id) => {
  return async (dispatch, getState) => {
    const {
      formats: { items },
    } = getState();

    if (items[id]) {
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${FORMAT_API}/${id}`,
      method: 'get',
    });
    dispatch(addFormat(response.data));

    dispatch(setLoading(false));
  };
};

export const setLoading = (loading) => {
  return {
    type: SET_FORMAT_LOADING,
    payload: { loading },
  };
};

export const addFormat = (format) => {
  return {
    type: ADD_FORMAT,
    payload: { format },
  };
};

export const addFormats = (formats) => {
  return {
    type: ADD_FORMATS,
    payload: {
      formats: buildObjectOfItems(formats),
    },
  };
};

export const setFormatRequest = (req, total) => {
  return {
    type: SET_FORMAT_REQUEST,
    payload: { req, total },
  };
};

export const setFormatIds = (ids) => {
  return {
    type: SET_FORMAT_IDS,
    payload: { ids },
  };
};

export const resetFormat = () => {
  return {
    type: RESET_FORMAT,
  };
};
