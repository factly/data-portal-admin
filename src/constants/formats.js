export const ADD_FORMAT = 'ADD_FORMAT';
export const ADD_FORMATS = 'ADD_FORMATS';
export const SET_FORMAT_LOADING = 'SET_FORMAT_LOADING';
export const SET_FORMAT_REQUEST = 'SET_FORMAT_REQUEST';
export const SET_FORMAT_IDS = 'SET_FORMAT_IDS';
export const RESET_FORMAT = 'RESET_FORMAT';

export const FORMAT_API = `/formats`;
