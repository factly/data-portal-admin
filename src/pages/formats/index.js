import React from 'react';
import FormatsList from './components/FormatsList';
import { Button } from 'antd';
import { Link } from 'react-router-dom';

const Formats = () => {
  return (
    <>
      <Link to={'/formats/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Format
        </Button>
      </Link>
      <FormatsList />
    </>
  );
};

export default Formats;
