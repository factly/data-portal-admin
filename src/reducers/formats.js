import produce from 'immer';
import {
  ADD_FORMAT,
  ADD_FORMATS,
  SET_FORMAT_LOADING,
  SET_FORMAT_REQUEST,
  SET_FORMAT_IDS,
  RESET_FORMAT,
} from '../constants/formats';

const initialState = {
  loading: true,
  ids: [],
  req: [],
  items: {},
  total: 0,
};

const formatsReducer = produce((draft, action = {}) => {
  switch (action.type) {
    case SET_FORMAT_LOADING:
      draft.loading = action.payload.loading;
      return;
    case ADD_FORMAT: {
      const { format } = action.payload;
      draft.items[format.id] = format;
      return;
    }
    case ADD_FORMATS: {
      const { formats } = action.payload;
      Object.assign(draft.items, formats);
      return;
    }
    case SET_FORMAT_IDS:
      draft.ids = action.payload.ids;
      return;
    case SET_FORMAT_REQUEST: {
      const { req, total } = action.payload;
      draft.req.push(req);
      draft.total = total;
      return;
    }
    case RESET_FORMAT:
      return initialState;
  }
}, initialState);

export default formatsReducer;
