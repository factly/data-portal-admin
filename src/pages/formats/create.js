import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import FormatCreateForm from './components/FormatCreateForm';
import { createFormat } from '../../actions/formats';

const FormatCreate = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const onCreate = (values) => {
    dispatch(createFormat(values));
    history.push('/formats');
  };

  return <FormatCreateForm onSubmit={onCreate} />;
};

export default FormatCreate;
