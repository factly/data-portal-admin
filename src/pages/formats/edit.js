import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import FormatCreateForm from './components/FormatCreateForm';
import { getFormat, updateFormat } from '../../actions/formats';

const EditFormat = () => {
  const { id } = useParams();
  const history = useHistory();
  const dispatch = useDispatch();
  const { format } = useSelector(({ formats }) => formats);

  React.useEffect(() => {
    dispatch(getFormat(id));
  }, []);

  if (!format.id) return <Skeleton />;

  const onUpdate = async (values) => {
    await dispatch(updateFormat(id, values));
    history.push('/formats');
  };

  return <FormatCreateForm data={format} onSubmit={onUpdate} />;
};

export default EditFormat;
