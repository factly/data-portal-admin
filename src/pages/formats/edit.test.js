import React from 'react';
import { useHistory } from 'react-router-dom';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { useDispatch, Provider, useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

import '../../matchMedia.mock';
import EditFormat from './edit';
import * as actions from '../../actions/formats';
import FormatEditForm from './components/FormatCreateForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn().mockReturnValue({ id: '1' }),
}));

jest.mock('../../actions/formats', () => ({
  getFormat: jest.fn(),
  addFormat: jest.fn(),
  updateFormat: jest.fn(),
}));

describe('Formats List component', () => {
  const store = mockStore({
    formats: {
      req: [],
      details: {
        '1': {
          id: 1,
          name: 'Format-1',
          slug: 'format-1',
          description: 'description',
        },
        '2': {
          id: 2,
          name: 'Format-2',
          slug: 'format-2',
          description: 'description',
        },
      },
      loading: true,
    },
  });
  store.dispatch = jest.fn(() => ({}));

  const mockedDispatch = jest.fn();
  useDispatch.mockReturnValue(mockedDispatch);

  describe('snapshot testing', () => {
    it('should render the component', () => {
      useSelector.mockReturnValueOnce({
        format: {
          id: 1,
          name: 'format',
          slug: 'slug',
          description: 'description',
        },
        loading: false,
      });
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditFormat />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with empty data', () => {
      useSelector.mockReturnValueOnce({
        format: {},
        loading: false,
      });
      let component;
      store.details = {};
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditFormat />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match skeleton while loading', () => {
      useSelector.mockReturnValueOnce({
        format: {},
        loading: true,
      });
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditFormat />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    afterEach(() => {
      wrapper.unmount();
    });
    it('should call get action', () => {
      useSelector.mockReturnValueOnce({ format: null, loading: true });
      actions.getFormat.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditFormat />
          </Provider>,
        );
      });
      expect(actions.getFormat).toHaveBeenCalledWith('1');
    });
    it('should call updateFormat', () => {
      useSelector.mockReturnValueOnce({ format: {}, loading: false });
      actions.updateFormat.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditFormat />
          </Provider>,
        );
      });
      wrapper.find(FormatEditForm).props().onSubmit({ test: 'test' });
      expect(actions.updateFormat).toHaveBeenCalledWith('1', { test: 'test' });
      expect(push).toHaveBeenCalledWith('/formats');
    });
  });
});
