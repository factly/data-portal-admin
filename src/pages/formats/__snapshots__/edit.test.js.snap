// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formats List component snapshot testing should match component with empty data 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="formats_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_name"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="formats_create_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. PDF"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="formats_create_description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Locked file"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_is_default"
        title="Is Default"
      >
        Is Default
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            aria-checked={true}
            className="ant-switch ant-switch-checked"
            id="formats_create_is_default"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <span
                aria-label="check"
                className="anticon anticon-check"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="check"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Formats List component snapshot testing should match skeleton while loading 1`] = `
<div
  className="ant-skeleton"
>
  <div
    className="ant-skeleton-content"
  >
    <h3
      className="ant-skeleton-title"
      style={
        Object {
          "width": "38%",
        }
      }
    />
    <ul
      className="ant-skeleton-paragraph"
    >
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": "61%",
          }
        }
      />
    </ul>
  </div>
</div>
`;

exports[`Formats List component snapshot testing should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="formats_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_name"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="formats_create_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. PDF"
            type="text"
            value="format"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="formats_create_description"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Locked file"
            type="text"
            value="description"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="formats_create_is_default"
        title="Is Default"
      >
        Is Default
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            aria-checked={true}
            className="ant-switch ant-switch-checked"
            id="formats_create_is_default"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <span
                aria-label="check"
                className="anticon anticon-check"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="check"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
